### ISSUE TITLE
Regional differences in number formats can cause OpenCAGE to destroy floats

### ISSUE THREAD

**Issue by mc68k** [MattFiler/OpenCAGE#80]
An issue was discovered with regional number formats that causes floats to be converted into integers.  If the system region number format is in the format of: xx,xxx.yy (thousands separator is a comma and decimal indicator is a period) there is no problem.  However, if the system region number format is: xx.xx,yy (thousand separator is a period and decimal indicator is a comma) floats get converted to integers like so:

30.1 becomes 301

This of course, breaks things.

Issue can be reproduced by setting the system number format to e.g. English (Sweden) and attempting to edit nodes in OpenCAGE containing floats.  Saving and reloading the file will have them as integer numbers instead of floats.  Trying to restore the float by entering it in the field throws an exception.

Resetting the region to e.g. English (United Kingdom) or English (United States) causes the issue to go away.

In summary, OpenCAGE doesn't seem to take into consideration number formats in different regions resulting in broken maps.

**Comment by MattFiler** [comment]
Good spot! I assume this is targetted specifically at the script editor?

**Comment by mc68k** [comment]
Yes, that's where the trouble lies.

**Comment by MattFiler** [comment]
Please could you provide screenshots of some examples where this occurs, and the entities/composites/levels that you've experienced it in? I'll get a fix up on staging once I can replicate it. Cheers!

**Comment by MattFiler** [comment]
I've just pushed an update out to the staging branch which should hopefully fix this issue. Within OpenCAGE's first screen, select the cog icon, tick "receive updates from staging branch" and click apply. You should receive the update, and be on version 0.9.2.84 when OpenCAGE reopens.

If this update hasn't fixed it, please could you provide screenshots of some examples where it occurs, and the entities/composites/levels that you've experienced it in?

**Comment by mc68k** [comment]
Appears to be fixed in 0.9.2.84.

**Comment by MattFiler** [comment]
Awesome, thanks!
